store.go

```diff
@@ -2666,6 +2706,11 @@ func (s *store) Mounted(id string) (int, error) {
 func (s *store) Unmount(id string, force bool) (bool, error) {
 	if layerID, err := s.ContainerLayerID(id); err == nil {
 		id = layerID
+	} else {
+		img, err := s.Image(id)
+		if err == nil {
+			id = img.TopLayer
+		}
```

Comment:
Can we add an `UnmountImage` instead? At the moment, the change would require a major version bump as the behaviour is backward incompatible.

Reply:
I originally did it that way, but I ended up duplicating a lot of code and made the callers more complex

I don't see this really as a modification of the API previously the caller would have gotten an error if they gave an image id to a umount function, now if they give the imageid, it will work. Having the Containers ID and The ImageID match would be just about impossible.

Reply:
>  as a modification of the API previously the caller would have gotten an error if they gave an image id to a umount function, now if they give the imageid, it will work. 

That is pretty much the definition of a breaking change: different results for the same input.

The functions could still share code if we introduce a third private one.